### PR TITLE
Update deliver policy docs

### PR DESCRIPTION
**Pull request by Jarema**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [ ] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [ ] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Issue reference
updates docs for PR: https://github.com/dapr/components-contrib/pull/2315

### CODE REVIEW

**Review by hhunter-ms — APPROVED** [review]
lgtm

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-jetstream.md [hunk 1]
```diff
@@ -97,6 +97,7 @@ spec:
 | rateLimit       |    N     | [Rate Limit]                               | `1024`                           |
 | hearbeat        |    N     | [Hearbeat]                                 | `10s`                            |
 | ackPolicy       |    N     | [Ack Policy]                               | `explicit`                       |
+| deliverPolicy   |    N     | One of: all, last, new, sequence, time     | `all`                            |
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-jetstream.md:99]:
```suggestion
| deliverPolicy   |    N     | One of: `all`, `last`, `new`, `sequence`, `time`     | `all`                            |
```

Comment (msfussell) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/setup-jetstream.md:99]: @Jarema - Can you also remove the "deliverAll" from the Spec Metadata table. This line.
deliverAll | N | Set deliver all as Replay Policy | true

Comment (Jarema) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/setup-jetstream.md:99]: Ah, sorry, missed that!
Removed.

**Pull request merged**